x-pack/plugins/threat_intelligence/public/components/indicator_table_cell_renderer/indicator_table_cell_renderer.tsx

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { unwrapValue } from '../../modules/indicators/lib/unwrap_value';
+import { Indicator, RawIndicatorFieldId } from '../../../common/types/indicator';
+import { EMPTY_VALUE } from '../../../common/constants';
+import { DateFormatter } from '../date_formatter';
+
+export interface IndicatorTableCellRendererFactoryProps {
+  indicator: Indicator;
+  field: string;
+  fieldTypesMap: { [id: string]: string };
+}
+
+export const indicatorTableCellRendererFactory = ({
```

Comment:
If I'm not missing smth this can be just a component, smth like that 
```
export const IndicatorField = ({
  indicator,
  field,
  fieldTypesMap,
}: IndicatorTableCellRendererFactoryProps) => {
  const value = unwrapValue(indicator, field as RawIndicatorFieldId);
  const fieldType = fieldTypesMap[field];
  return fieldType === 'date' ? (
    <DateFormatter date={value as string} />
  ) : (
    <>{value}</> || <>{EMPTY_VALUE}</>
  );
};
```
I think it's more natural for React than a factory. Not sure if the name `IndicatorField` is good, but I'd make it decoupled from "table cell", because in the end we just render a field value of an indicator, not necessarily in a table context

Reply:
@maxcold I really like this approach, thanks for suggesting it! [This commit](https://github.com/elastic/kibana/pull/138106/commits/92b55743b60d4dd6ee4eb2ea49a5eb8e7df1f34f) implements everything, including tests and storybook